### PR TITLE
feat: improve flashing speed

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
This PR improves the flashing speed by ignoring the `pollTimeout` returned by ST's DFU bootloader, which is way too long and just polls the status continuously.

### CODE REVIEW

File: src/shared/dfu/index.ts [hunk 1]
```diff
@@ -635,9 +635,12 @@ export class WebDFU {
       (ignoreErrors || dfuStatus.state !== dfuCommands.dfuERROR)
     ) {
       // eslint-disable-next-line no-await-in-loop
-      await asyncSleep(dfuStatus.pollTimeout);
-      // eslint-disable-next-line no-await-in-loop
-      dfuStatus = await this.getStatus();
+      await asyncSleep(1);
```
Comment (freshollie) [inline src/shared/dfu/index.ts:638]: Is this arbitrary?

Comment (raphaelcoeffic) [reply src/shared/dfu/index.ts:638]: The goal is to avoid clogging the worker with real busy wait.

**Comment by pfeerick** [comment]
Writing EdgeTX 2.10.3 to TX16S on Windows:

With https://6c6314e7.buddy-7a6.pages.dev
55s Overall
23s Erase
32s Write

With buddy.edgetx.org
55s Overall
24s Erase
31s Write

Taking into account rounding error, and observer error/lag, basically no difference here atm yet. 

`dfu-util` is a hell of a lot slower... 
still around 24s to erase, but 2m39s - 3m02s overall - to DFU flash same firmware (as configured by Companion).

File: src/shared/dfu/index.ts (hunk 1 above)
Comment (freshollie) [reply src/shared/dfu/index.ts:638]: Ah! If that's the reasoning, I think `asyncSleep(0)` also works, as `setTimeout` always runs on the next event loop execution

Comment (raphaelcoeffic) [reply src/shared/dfu/index.ts:638]: Nice, thx! I'll try it out then!

**Comment by raphaelcoeffic** [comment]
> Taking into account rounding error, and observer error/lag, basically no difference here atm yet.

It seems the PR has been deployed to `buddy.edgetx.org` as well...

Screenshot 2024-08-20 at 17 19 07

File: src/shared/dfu/index.ts (hunk 1 above)
Comment (raphaelcoeffic) [reply src/shared/dfu/index.ts]: Done, seems to work properly, thx!

**Comment by pfeerick** [comment]
Ok, that was more like it... Buddy now took about 24s to erase, 2m37s to write, for a total of 3m01s (on Linux) vs the (still) 55s overall with the preview build... T12MAX was ~18-19 seconds to erase and flash... at these speeds... I'll be DFU flashing instead of uploading to SD and bootloader flashing from now on :rofl:

**Comment by freshollie** [comment]
Nice! Not sure what configuration has been made for the pages to deploy to production 🤔

**Review by freshollie — APPROVED** [review]

**Review by pfeerick — APPROVED** [review]
Given I have been flashing with [pyfu-usb](https://pypi.org/project/pyfu-usb/) in this way for over a year now for all the various targets... I consider this DFU timing change (from dumb timeout to polling) stable/reliable for production... I have had no issues to date arising from it ;) 

edit: further, and more relevantly, I have now been using the preview version of this for flashing various handsets and have had no issues at all.

**Pull request merged**